src/overlays/actors/ovl_Obj_Spinyroll/z_obj_spinyroll.h

```diff
@@ -6,12 +6,54 @@
 struct ObjSpinyroll;
 
 typedef void (*ObjSpinyrollActionFunc)(struct ObjSpinyroll*, GlobalContext*);
+typedef s32 (*ObjSpinyrollColFunc)(CollisionContext*, f32*, CollisionPoly**, s32*, Vec3f*, struct ObjSpinyroll*);
+
+#define OBJSPINYROLL_GET_7F(thisx) ((thisx)->params & 0x7F)
+#define OBJSPINYROLL_GET_380(thisx) (((thisx)->params >> 7) & 7)
+#define OBJSPINYROLL_GET_1C00(thisx) (((thisx)->params >> 0xA) & 7)
+#define OBJSPINYROLL_GET_C000(thisx) (((thisx)->params >> 0xE) & 3)
+
+#define OBJSPINYROLL_GET_7F_7F 0x7F
+#define OBJSPINYROLL_GET_C000_0 0
+#define OBJSPINYROLL_GET_C000_1 1
+
+typedef struct {
+    /* 0x00 */ f32 unk_00;
+    /* 0x04 */ CollisionPoly* unk_04;
+    /* 0x08 */ s32 bgId;
+} ObjSpinyrollSubStruct; // size = 0xC
+
+typedef struct {
+    /* 0x00 */ ObjSpinyrollSubStruct unk_00[10];
+} ObjSpinyrollStruct; // size = 0x78
+
+typedef struct {
+    /* 0x00 */ ObjSpinyrollStruct unk_00[2];
+    /* 0xF0 */ ObjSpinyrollStruct* unk_F0;
+    /* 0xF4 */ s32 unk_F4;
+} ObjSpinyrollStruct2; // size = 0xF8?
 
 typedef struct ObjSpinyroll {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x35C];
+    /* 0x0000 */ DynaPolyActor dyna;
+    /* 0x015C */ ColliderTris collider;
+    /* 0x017C */ ColliderTrisElement colliderElements[6];
+    /* 0x03A4 */ ObjSpinyrollStruct2 unk_3A4;
+    ///* 0x0494 */ ObjSpinyrollStruct* unk_494;
+    ///* 0x0498 */ s32 unk_498;
```

Comment:
I'm assuming these commented-out structs are to help with the non-matchings?

Reply:
Pretty much, because I'm not sure on how the layout is meant to be, just some other ideas when triyng to match.

Reply:
In that case, could you add a brief comment on one of these lines. something simple like `// possible alternative that may be used for non-matchings`